philosowaffle/peloton-to-garmin
# Add support for GPS data from outdoor activities - #35

Closes #35 

Summary of changes:

- Updates to tcx_builder.py
  - Flatten data from Peloton API to return a dict keyed on the second_offset_from_start
  - Add function to convert feet to meters if needed
  - Map data and add relevant TCX elements
- Updates to tests
   - Add randomized location_data to peloton_workoutsamples_running_outdoor
   - Add test in tcx_test_helper.py to check position data is present and correct
   - Plumb test into test_tcx_builder.py

E2E Tests:
- Converted updated outdoor run to TCX and uploaded to Strava - worked as expected
- Converted personal run to TCX and uploaded to Strava - verified that position/elevation data matches route I ran

 Test output:
```
$ pytest
=================================================================================== test session starts ====================================================================================
platform darwin -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /Users/[removed]/code/peloton-to-garmin/tests
plugins: mock-3.5.1
collected 29 items                                                                                                                                                                         

test_garminClient.py ........                                                                                                                                                        [ 27%]
test_peloton-to-garmin.py .                                                                                                                                                          [ 31%]
test_tcx_builder.py ....................                                                                                                                                             [100%]

==================================================================================== 29 passed in 0.99s ====================================================================================
```
